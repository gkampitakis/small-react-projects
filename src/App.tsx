import React from 'react';
import Birthday from './projects/birthday/birthday';
import Tours from './projects/tours/tours';

function App () {
  return (
    <div className="App">
      <Tours />
    </div>
  );
}

export default App;
