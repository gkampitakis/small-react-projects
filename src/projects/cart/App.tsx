import React, { ReactElement } from 'react';
import { useGlobalContext } from './context';
import Navbar from './components/Navbar';
import Cart from './components/Cart';

export default function App (): ReactElement {
  const { loading } = useGlobalContext();

  return (<main className="cart_project">
    {loading &&
      <div className="loading">
        <h1>Loading ...</h1>
      </div>
    }
    {!loading && <>
      <Navbar />
      <Cart />
    </>}
  </main>);
}