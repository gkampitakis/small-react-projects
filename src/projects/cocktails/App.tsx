import React, { ReactElement, lazy, Suspense } from 'react';
import { Route, Switch, } from 'react-router-dom';
import { useBodyStyles } from '../../hooks';
import { AppProvider } from './context';
import Loading from '../../components/Loading';
import Home from './pages/Home';
import Navbar from './components/Navbar';
import './index.scss';


export default function App (): ReactElement {
  useBodyStyles({
    background: '#f1f5f8'
  });

  return (
    <AppProvider>
      <main className="cocktails_project">
        <Navbar />
        <Switch>
          <Route exact component={Home} path="/cocktails" />
          <Route exact component={() => LazyDetails('About')} path="/cocktails/about" />
          <Route exact component={() => LazyDetails('Details')} path="/cocktails/details/:id" />
        </Switch>
      </main>
    </AppProvider>
  );
}

const LazyDetails = (page: string) => {
  const Component = lazy(() => import(`./pages/${page}`));

  return (
    <Suspense fallback={<Loading />}><Component /></Suspense >
  );
}
