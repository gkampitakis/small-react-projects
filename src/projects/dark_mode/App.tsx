import React, { ReactElement, useEffect } from 'react';
import { useBodyStyles, useDocTitle, useLocalStorage } from '../../hooks';
import Article from './Article';
import data from './data';
import './index.scss';

export default function App(): ReactElement {
  useDocTitle('Dark Mode');
  useBodyStyles({
    color: 'var(--clr-font)',
    background: 'var(--clr-bcg)'
  });
  const [theme, setTheme] = useLocalStorage<string>('dark-mode', 'dark_theme');

  useEffect(() => {
    document.documentElement.className = theme;
  }, [theme]);

  function toggleTheme() {
    const themes: Record<string, string> = {
      dark_theme: 'light_theme',
      light_theme: 'dark_theme'
    };

    setTheme(themes[theme]);
  }

  return (
    <main className="dark_mode_project">
      <header>
        <div>
          <h1>overreacted</h1>
          <button onClick={toggleTheme}>toggle</button>
        </div>
      </header>
      <section className="articles">
        {data.map(({ id, ...rest }) => (
          <Article key={id} {...rest} />
        ))}
      </section>
    </main>
  );
}
