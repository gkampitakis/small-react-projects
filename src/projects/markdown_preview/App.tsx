import React, { ReactElement, useState } from 'react';
import ReactMarkdown from 'react-markdown';
import { useBodyStyles } from '../../hooks';
import 'github-markdown-css';
import './index.scss'

export default function App (): ReactElement {
  useBodyStyles({
    background: 'hsl(210, 36%, 96%)'
  });
  const [preview, setPreview] = useState('# markdown preview');

  return (
    <main className="markdown_preview">
      <section>
        <textarea
          value={preview}
          onChange={(e) => setPreview(e.target.value)}
        />
        <article>
          <ReactMarkdown className="markdown-body">{preview}</ReactMarkdown>
        </article>
      </section>
    </main>
  );
}
