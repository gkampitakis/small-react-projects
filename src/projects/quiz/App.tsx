import React, { ReactElement } from 'react';
import { AppProvider } from './context';
import Quiz from './Components/Quiz';
import './index.scss';
import { useBodyStyles, useDocTitle } from '../../hooks';

export default function App(): ReactElement {
  useBodyStyles({
    background: 'hsl(210, 36%, 96%)'
  });
  useDocTitle('Quiz');

  return (
    <AppProvider>
      <Quiz />
    </AppProvider>
  );
}
