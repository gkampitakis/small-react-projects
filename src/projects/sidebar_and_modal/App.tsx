import React from 'react';
import Home from './components/Home';
import Modal from './components/Modal';
import Sidebar from './components/Sidebar';
import { AppProvider } from './context';
import './index.css';

export default function App () {
  return (
    <AppProvider>
      <Home />
      <Modal />
      <Sidebar />
    </AppProvider>
  )
}
