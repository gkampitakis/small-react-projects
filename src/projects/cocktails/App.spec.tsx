import React from 'react';
import { FetchMock } from 'jest-fetch-mock/types';
import { BrowserRouter } from 'react-router-dom';
import { render, act } from '@testing-library/react';
import App from './App';

const fetchMock = (fetch as FetchMock);
const mockData = [
  {
    idDrink: 0,
    strDrinkThumb: 'mockImg0',
    strAlcoholic: 'mockInfo0',
    strGlass: 'mockGlass0'
  },
  {
    idDrink: 1,
    strDrinkThumb: 'mockImg1',
    strAlcoholic: 'mockInfo1',
    strGlass: 'mockGlass1'
  },
  {
    idDrink: 2,
    strDrinkThumb: 'mockImg2',
    strAlcoholic: 'mockInfo2',
    strGlass: 'mockGlass2'
  },
  {
    idDrink: 3,
    strDrinkThumb: 'mockImg3',
    strAlcoholic: 'mockInfo3',
    strGlass: 'mockGlass3'
  }
];

describe('Cocktail', () => {

  beforeAll(fetchMock.enableMocks);
  beforeEach(() => {
    fetchMock.resetMocks();
    fetchMock.mockResponseOnce(JSON.stringify(mockData));
  });

  it('Should render correctly', async () => {
    await act(async () => {
      const { baseElement } = render(<App />,
        { wrapper: BrowserRouter });

      expect(baseElement).toMatchSnapshot();
    });
  });
});
