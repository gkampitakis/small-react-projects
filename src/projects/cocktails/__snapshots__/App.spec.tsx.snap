// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail Should render correctly 1`] = `
<body>
  <div>
    <main
      class="cocktails_project"
    >
      <nav>
        <div
          class="nav-center"
        >
          <a
            href="/cocktails"
          >
            <img
              alt="Cocktail Logo"
              src="logo.svg"
            />
          </a>
          <ul>
            <li>
              <a
                href="/cocktails"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/cocktails/about"
              >
                About
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </main>
  </div>
</body>
`;
